x/data-proxy/keeper/msg_server.go

```diff
@@ -60,11 +60,13 @@ func (m msgServer) RegisterDataProxy(goCtx context.Context, msg *types.MsgRegist
 	}
 
 	feeBytes := []byte(msg.Fee.String())
+	adminAddressBytes := []byte(msg.AdminAddress)
 	payoutAddressBytes := []byte(msg.PayoutAddress)
 	memoBytes := []byte(msg.Memo)
 
 	payload := make([]byte, 0, len(feeBytes)+len(payoutAddressBytes)+len(memoBytes))
```

Comment:
Perhaps we can change the capacity?

Reply:
Not sure how to accomplish that without making the code dirty..

Reply:
Good catch! I might just move the sum out but it'll still be kinda dirty :p

Reply:
I also added memo length validation to the `ValidateBasic` methods.

Reply:
LGTM!